lib/linter/linter.js

```diff
@@ -935,9 +939,12 @@ function createRuleListeners(rule, ruleContext) {
  * @param {string | undefined} cwd cwd of the cli
  * @param {string} physicalFilename The full path of the file on disk without any code block information
  * @param {Function} ruleFilter A predicate function to filter which rules should be executed.
+ * @param {boolean} stats If true, stats are collected appended to the result
+ * @param {WeakMap<Linter, LinterInternalSlots>} slots InternalSlotsMap of linter
  * @returns {LintMessage[]} An array of reported problems
  */
-function runRules(sourceCode, configuredRules, ruleMapper, parserName, languageOptions, settings, filename, disableFixes, cwd, physicalFilename, ruleFilter) {
+function runRules(sourceCode, configuredRules, ruleMapper, parserName, languageOptions, settings, filename, disableFixes, cwd, physicalFilename, ruleFilter,
+    stats, slots) {
```

Comment:
It looks like we end up passing `slots` around unnecessarily (from here into timing.js and into stats.js). Can we just pull out the necessary values and pass them along?

(`slots` is really not supposed to be shared outside of `Linter`.)

Reply:
Yes, this seems a bit redundant now that you mention it. ☺️ I've just updated the code accordingly:
- Removed `slots` from _timing.js_ `time()`
- Moved `storeTime()` to the Linter, since:
  - It is not used anywhere else
  - It's job is to store/update the time delta/ fixPasses in the Linter's slots map
  - Hence, `slots` is ok to use as input here (as we are inside the Linter).